contracts/RootChain/RootChain.sol

```diff
@@ -187,20 +198,41 @@ contract RootChain {
             require(merkleHash.checkMembership(txPos[1], childChain[txPos[0]].root, proof));
         }
 
-        // one-to-one mapping between priority and exit
-        require(exits[priority].owner == address(0));
-        require(exits[priority].amount == 0);
+        // check that the UTXO's two direct inputs have not been previously exited
+        validateExitInputs(txList);
 
+        // create new started/pending exit after passing all previous checks
         exitsQueue.insert(priority);
-
         exits[priority] = exit({
-            owner: txList[10 + 2 * txPos[2]].toAddress(),
             amount: txList[11 + 2 * txPos[2]].toUint(),
             utxoPos: txPos,
-            created_at: block.timestamp
+            created_at: block.timestamp,
+            owner: txList[10 + 2 * txPos[2]].toAddress(),
+            state: 1,
+            sigs: sigs
         });
     }
 
+    /// For any attempted exit of an UTXO, validate that the UTXO's two inputs have not
+    /// been previously exited. If UTXO's inputs are in the exit queue, those inputs'
+    /// exits are deleted from the exit queue and the current UTXO's exit remains valid.
```

Comment:
Update comments. If inputs are in exit queue, those inputs must be challenged in order for the direct descendants to start a valid exit